test/regression/riscv/instructions/float.cc

```diff
@@ -648,6 +771,32 @@ TEST_P(InstFloat, FCVT_WU_S) {
   EXPECT_EQ(getGeneralRegister<uint64_t>(29), 0);
   EXPECT_EQ(getGeneralRegister<uint64_t>(7), 0xFFFFFFFFFFFFFFFF);
   EXPECT_EQ(getGeneralRegister<uint64_t>(30), 0xFFFFFFFFFFFFFFFF);
+
+  initialHeapData_.resize(32);
+  uint64_t* intHeap = reinterpret_cast<uint64_t*>(initialHeapData_.data());
+  intHeap[0] = 0xffffffff3f800000;  // Correct NaN boxing of no. 1
+  intHeap[1] = 0xfff7ffff3f800000;  // Incorrect NaN boxing of no. 1
+
+  RUN_RISCV(R"(
+    # Get heap address
+    li a7, 214
+    ecall
+
+    fld fa3, 0(a0) # Correct
+    fld fa4, 8(a0) # Incorrect
+
+    fcvt.wu.s t0, fa3 # fcvt 1 = 1
+    fcvt.wu.s t1, fa4 # Incorrect NaN box should be caught by fcvt and
+                          # canonical NaN used as input, fcvt NaN = 2^32 − 1
+)");
+
+  EXPECT_EQ(getFPRegister<uint64_t>(13), 0xffffffff3f800000);
+  EXPECT_EQ(getFPRegister<uint64_t>(14), 0xfff7ffff3f800000);
+
+  EXPECT_EQ(getGeneralRegister<uint64_t>(5), 0x0000000000000001);
+  EXPECT_EQ(getGeneralRegister<uint32_t>(6),
+            (uint64_t)pow(2, 32) - 1);  // Expected result from spec
+  EXPECT_EQ(getGeneralRegister<uint64_t>(6), 0xFFFFFFFFFFFFFFFF);
```

Comment:
Why do you check register 6 twice; once as uint32 and once as uint64?

Reply:
Strictly speaking only the hex check is necessary. The cast shouldn't be there so removed. The instructions produces a result of size word (32 bits) and the spec describes this as being the value 2^32-1 so this is what the first check is for, this is a bit more human readable and easier to align with the spec. The result should also be sign extended, so the full 64 bits are checked against the hex. I could reduce this to one check and describe the rest in a short comment

Reply:
Two checks is fine, was just curious. Removing the cast though was needed yeah